docs/extending/extending_client_side.md

```diff
@@ -0,0 +1,528 @@
+
+In a similar way, Wagtail's [template components](template_components) provide a `media` property or `Media` class to add scripts when rendered.
+
+These will ensure the added files are used in the admin after the core JavaScript admin files are already loaded.
+
+(extending_client_side_using_events)=
+
+## Extending with DOM events
+
+When approaching client-side customisations or adopting new components, try to keep the implementation simple first, you may not need any knowledge of Stimulus, React, JavaScript Modules or a build system to achieve your goals.
+
+The simplest way to attach behaviour to the browser is via DOM Events and plain (vanilla) JavaScript.
+
+For example, if you want to attach some logic to a field value change in Wagtail you can add an event listener, check if it's the correct element and change what you need.
+
+```javascript
+// myapp/static/js/events.js
+document.addEventListener('change', function (event) {
+    if (event.target) {
+        console.log('field has changed', event.target);
+    }
+});
+```
+
+Alternatively, you can write JavaScript logic to achieve this by listening to all click events and checking if the clicked element corresponds to the comments section.
+
+```javascript
+// myapp/static/js/events.js
+document.addEventListener('click', function (event) {
+    if (!event.target) return;
+
+    isCommentsToggle = event.target.matches(
+        '[data-side-panel-toggle="comments"]',
+    );
+
+    if (!isCommentsToggle) return;
+    console.log('Comments have been toggled');
+});
+```
+
+Then, you can register this with the `insert_global_admin_js` hook.
+
+```python
+# myapp/wagtail_hooks.py
+from django.templatetags.static import static
+from django.utils.safestring import mark_safe
+
+from wagtail import hooks
+
+@hooks.register('insert_global_admin_js')
+def global_admin_js():
+    return mark_safe(
+        f'<script src="{static("js/events.js")}"></script>',
+    )
+```
+
+### Wagtail's custom DOM events
+
+-   Wagtail supports some custom behaviour via listening or dispatching custom DOM events, usually with the prefix `wagtail:` or `w-` for specific Stimulus controllers.
+-   See [Images title generation on upload](images_title_generation_on_upload).
+-   See [Documents title generation on upload](docs_title_generation_on_upload).
+
+(extending_client_side_stimulus)=
+
+## Extending with Stimulus
+
+Wagtail uses [Stimulus](https://stimulus.hotwired.dev/) as a way to provide lightweight client-side interactivity where React isn't required.
+You can use Stimulus to easily build custom JavaScript widgets within the admin interface.
+The key benefit of using Stimulus is that your code can avoid the need for manual initialisation when widgets appear dynamically in the browser. Such as within modals, `InlinePanels` or `StreamField` panels.
+
+### Understanding the basics of Stimulus
```

Comment:
I don’t get why we have all of this in our docs. Wouldn’t people be better off by just looking at the Stimulus docs?

Reply:
Yeah, I struggled with this. I found myself referencing Stimulus concepts later and it felt like I should have some basics on this page.

My reasoning is the audience could get away with getting something custom working and understood without reading an external page. 

Happy to remove this completely and just direct people to the Stimulus docs.